### PR TITLE
add more buttons; remove some sections covered in the docs; general cleanup

### PR DESCRIPTION
**Pull request by RubenKelevra**
- added dynamic buttons for packages/downloads which shows the latest version where possible
- dynamic next milestone buttons
- reordered to have package managers and docker on top, manual downloads afterwards
- remove the docker and the cli section, since they are covered equally in the docs + button links
- added more sections to the navigation, to find stuff easier
- dynamic go requirement version button
- some social media buttons
- general cleanup

### CODE REVIEW

File: README.md
Comment (Stebalien) [inline README.md:30]: I like this, but it'll get out of date pretty quickly. It's probably safer to just link to the milestone list itself (and forgo the badges).

File: README.md
Comment (Stebalien) [inline README.md:86]: macOS is, unfortunately, the "correct" capitalization (according to apple).

File: README.md
Comment (Stebalien) [inline README.md:165]: There's no reason to remove the instructions.

File: README.md
Comment (Stebalien) [inline README.md:191]: Wow. This is GREAT! (we're terrible at updating the readme to reflect the latest version)

File: README.md
Comment (Stebalien) [inline README.md:108]: I wouldn't use the AUR badge here. Git PKGBUILDs are rarely updated, so the version will displayed here fall behind.

File: README.md
Comment (Stebalien) [inline README.md:84]: This is a normal linux package manager.

Really, instead of splitting these up, I'd keep them all in the "linux package managers" section.

I get that you're trying to indicate that you can use `nix` on macOS, but it would make more sense to just have a macOS install section that links back to `nix` (which is _primarily_ used on Linux anyways).

File: README.md
Comment (Stebalien) [inline README.md:186]: This is a bit of a gratuitous use of shields. I'd just turn "GitHub releases" into a link.

File: README.md
Comment (Stebalien) [inline README.md:299]: The badge sizes are inconsistent. In general, I'd go for the smaller size.

File: README.md
Comment (Stebalien) [inline README.md:279]: :+1:

**Review by Stebalien — COMMENTED** [review]
Thanks! The readme is in desperate need of an overhaul.

**Comment by BigLep** [comment]
@RubenKelevra : can you please incorporate the feedback?  We'll then get this merged.

File: README.md
Comment (RubenKelevra) [reply README.md:30]: How about just adding the "remove badge for this milestone" on the release-todo list? I really like these ones :)

File: README.md
Comment (RubenKelevra) [reply README.md:86]: Ah okay :D

File: README.md
Comment (Stebalien) [reply README.md:30]: The release issue template is already massive. I'll defer to @aschmahmann here as he's the one following it now, but adding an "edit the readme" step to the already very long list seems annoying.

Comment (RubenKelevra) [reply README.md:30]: Okay, so we just wait for a decision from @aschmahmann

File: README.md
Comment (RubenKelevra) [reply README.md:165]: Well, you can visit the site and copy the instruction from it. I thought that's favorable for users :)

File: README.md
Comment (RubenKelevra) [reply README.md:108]: Well, I'm the maintainer. I can update it from time to time :)

File: README.md
Comment (RubenKelevra) [reply README.md:84]: Okay

File: README.md
Comment (RubenKelevra) [reply README.md:186]: Oh okay, I thought including the latest version would look fancy :)

I'll remove it.

File: README.md
Comment (RubenKelevra) [reply README.md]: Okay I reduced it.

**Comment by RubenKelevra** [comment]
@BigLep wrote:
> @RubenKelevra : can you please incorporate the feedback? We'll then get this merged.

Yeah sorry looks like I overlooked the notification :)

File: README.md
Comment (RubenKelevra) [reply README.md]: It also has the advantage to show the disclaimer for package security and provide further help. I think that's better than showing a random command string which feels a bit like `curl | sh`

Comment (Stebalien) [reply README.md]: Please put it back. It's nothing like `curl | sh`, it's a nice bit of documentation like every other install command included in the readme.

Comment (RubenKelevra) [reply README.md]: okay

**Comment by RubenKelevra** [comment]
@aschmahmann can do give this a quick look, please? :)

**Comment by BigLep** [comment]
@Stebalien : are you ok to close this one out since you have been involved in the incremental diffs?

**Comment by RubenKelevra** [comment]
Well, the only unchanged thing is the Milestones as a button. I mean, we could merge it and @aschmahmann can remove them if he feels like that's too much work :)

File: README.md
Comment (aschmahmann) [reply README.md]: I'm with @Stebalien here. We're trying to move towards automating as much of the release process as possible so we can roll out updates more incrementally. One more step in a list we're trying to condense towards "hit the button" isn't great at the moment.

Perhaps as that checklist shrinks we can add an Actions workflow updating the readme to point at various things about the current + next release, but until then let's just keep it simple.

**Review by Stebalien — APPROVED** [review]
Thanks!

**Comment by Stebalien** [comment]
We'll have to merge after https://github.com/ipfs/go-ipfs/pull/8401.

**Pull request merged**
